zipline/data/_adjustments.pyx

```diff
@@ -136,60 +145,85 @@ cdef _adjustments(object adjustments_db,
     while dividends_to_query:
         query_len = min(len(dividends_to_query), SQLITE_MAX_IN_STATEMENT)
         query_assets = dividends_to_query[:query_len]
-        t= [str(a) for a in query_assets]
-        statement = ADJ_QUERY_TEMPLATE.format('dividends',
-            ",".join(['?' for _ in query_assets]), start_date, end_date)
+        t = [str(a) for a in query_assets]
+        statement = ADJ_QUERY_TEMPLATE.format(
+            'dividends',
+            ",".join(['?' for _ in query_assets]),
+            start_date,
+            end_date,
+        )
         c.execute(statement, t)
         dividends_to_query = dividends_to_query[query_len:]
         dividends_results.extend(c.fetchall())
 
     return splits_results, mergers_results, dividends_results
 
 
-cpdef load_adjustments_from_sqlite(object adjustments_db,  # sqlite3.Connection
-                                   list columns,
+cpdef load_adjustments_from_sqlite(object adjustments_db,
                                    DatetimeIndex_t dates,
-                                   Int64Index_t assets):
+                                   Int64Index_t assets,
+                                   bool should_include_splits,
+                                   bool should_include_mergers,
+                                   bool should_include_dividends):
     """
-    Load a dictionary of Adjustment objects from adjustments_db
+    Load a dictionary of Adjustment objects from adjustments_db.
 
     Parameters
     ----------
     adjustments_db : sqlite3.Connection
         Connection to a sqlite3 table in the format written by
         SQLiteAdjustmentWriter.
-    columns : list[str]
-        List of column names for which adjustments are needed.
     dates : pd.DatetimeIndex
-        Dates for which adjustments are needed
+        Dates for which adjustments are needed.
     assets : pd.Int64Index
         Assets for which adjustments are needed.
+    should_include_splits : bool
+        Whether split adjustments should be included.
+    should_include_mergers : bool
+        Whether merger adjustments should be included.
+    should_include_dividends : bool
+        Whether dividend adjustments should be included.
 
     Returns
     -------
-    adjustments : list[dict[int -> Adjustment]]
-        A list of mappings from index to adjustment objects to apply at that
-        index.
+    adjustments : tuple[dict[int -> Adjustment]]
```

Comment:
Can we add a flag to control whether this returns price-type, volume-type, or both? If we load all columns except for `volume`, we don't need to construct this dict.